hooks/wallet/TokensAPI.ts

```diff
@@ -0,0 +1,46 @@
+import { ApiPagedResponse, WhaleApiClient } from '@defichain/whale-api-client'
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Dispatch } from 'redux'
+import { RootState } from '../../store'
+import { setTokens, setUtxoBalance, tokensSelector } from '../../store/wallet'
+import { useWhaleApiClient } from '../api/useWhaleApiClient'
+import { useWalletAPI } from './WalletAPI'
+
+const fetchWalletTokens = async (address: string, whaleAPI: WhaleApiClient): Promise<ApiPagedResponse<AddressToken>> => {
+  return await whaleAPI.address.listToken(address)
+}
+
+const fetchWalletBalance = async (address: string, whaleAPI: WhaleApiClient): Promise<string> => {
+  return await whaleAPI.address.getBalance(address)
+}
+
+const fetchWalletData = (address: string, dispatch: Dispatch<any>, whaleAPI: WhaleApiClient): void => {
+  fetchWalletTokens(address, whaleAPI).then((walletTokens) => {
+    dispatch(setTokens(walletTokens))
+  }).catch((error) => console.log(error))
+
+  fetchWalletBalance(address, whaleAPI).then((walletBalance) => {
+    dispatch(setUtxoBalance(walletBalance))
+  }).catch((error) => console.log(error))
+}
+
+export function useTokensAPI (): AddressToken[] {
+  const tokens = useSelector((state: RootState) => tokensSelector(state.wallet))
+  const account = useWalletAPI().getWallet().get(0)
+  const dispatch = useDispatch()
+  const whaleAPI = useWhaleApiClient()
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timeout
+    account.getAddress().then(async address => {
+      await fetchWalletData(address, dispatch, whaleAPI)
+      intervalId = setInterval(() => {
+        fetchWalletData(address, dispatch, whaleAPI)
+      }, 10000)
+    }).catch((error) => console.log(error))
+    return () => clearInterval(intervalId)
+  }, [])
```

Comment:
We have to drop the interval refreshing, at least for now, how tabs are implemented native means this will keep refreshing even if we have navigated into a deep controller.

This will reduce the complexity greatly for our code.

We will also need:
1. ability to trigger a manual refresh
2. pagination of all tokens

> Kinda awkward that this code section is here, would make more sense in redux slice but I think it is limited by `useWhaleApiClient` being a hook. In the future, we will have to refactor that for code simplicity later.

Reply:
For the interval part, I think we should keep it. The reason is users tend to wait on the page when a transaction is pending. Having to tell them to manual refresh or switch tabs might not be a good solution. It might be reported as a bug/issue in the future.

Reply:
I think you misunderstand what I mean, not saying we don't need it in the future but saying we should remove it now to reduce complexity and code debt.

For reasons that we need it updated, we can trigger the updates manually.

Reply:
I have remove the interval for tokens